correctness: fix seed 3153055325

Scheduled correctness found a issue:

```
+ ./bindings/bindingtester/bindingtester.py --num-ops 1000 --api-version 630 --test-name api --concurrency 5 rust

Creating test at API version 630
Generating api test at seed 3153055325 with 1000 op(s) and 5 concurrent tester(s)...

Inserting test into database...

Running tester '/home/runner/work/foundationdb-rs/foundationdb-rs/target/debug/bindingtester test_spec 630'...
thread 'b"thread\x5fspec3"' panicked at 'attempt to subtract with overflow', foundationdb-bindingtester/src/main.rs:718:72
stack backtrace:
   0: rust_begin_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:107:14
   2: core::panicking::panic
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:48:5
   3: bindingtester::StackMachine::run_step::{{closure}}
             at /home/runner/work/foundationdb-rs/foundationdb-rs/foundationdb-bindingtester/src/main.rs:718:72
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
   5: bindingtester::StackMachine::run::{{closure}}
             at /home/runner/work/foundationdb-rs/foundationdb-rs/foundationdb-bindingtester/src/main.rs:1578:21
   6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
   7: futures_executor::local_pool::block_on::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.19/src/local_pool.rs:315:23
   8: futures_executor::local_pool::run_executor::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.19/src/local_pool.rs:90:37
   9: std::thread::local::LocalKey<T>::try_with
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/local.rs:399:16
  10: std::thread::local::LocalKey<T>::with
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/local.rs:375:9
  11: futures_executor::local_pool::run_executor
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.19/src/local_pool.rs:86:5
  12: futures_executor::local_pool::block_on
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.19/src/local_pool.rs:315:5
  13: bindingtester::StackMachine::run_step::{{closure}}::{{closure}}
             at /home/runner/work/foundationdb-rs/foundationdb-rs/foundationdb-bindingtester/src/main.rs:1454:29
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'joined thread to not panic: Any { .. }', foundationdb-bindingtester/src/main.rs:1586:27
stack backtrace:
   0: rust_begin_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:107:14
   2: core::result::unwrap_failed
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/result.rs:1613:5
   3: core::result::Result<T,E>::expect
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/result.rs:1255:23
   4: bindingtester::StackMachine::join
             at /home/runner/work/foundationdb-rs/foundationdb-rs/foundationdb-bindingtester/src/main.rs:1586:13
   5: bindingtester::main
             at /home/runner/work/foundationdb-rs/foundationdb-rs/foundationdb-bindingtester/src/main.rs:1638:5
   6: core::ops::function::FnOnce::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

'/home/runner/work/foundationdb-rs/foundationdb-rs/target/debug/bindingtester' did not complete succesfully (exit code: 101)

Test of type api failed to complete successfully with random seed 3153055325 and 1000 operations

Error: Process completed with exit code 2.
```

Source: https://github.com/foundationdb-rs/foundationdb-rs/runs/4811486807?check_suite_focus=true#step:7:6949

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -212,7 +212,7 @@ jobs:
         run: scripts/setup_bindingtester.sh target/x86_64-unknown-linux-gnu/debug/bindingtester
 
       - name: Run bindingtester
-        run: scripts/run_bindingtester.sh 10
+        run: scripts/run_bindingtester.sh 50
 
       - id: coverage
         uses: actions-rs/grcov@v0.1

--- a/foundationdb-bindingtester/src/main.rs
+++ b/foundationdb-bindingtester/src/main.rs
@@ -17,6 +17,7 @@ use fdb::tuple::{pack, pack_into, unpack, Bytes, Element, Subspace, TuplePack};
 use fdb::*;
 use futures::future;
 use futures::prelude::*;
+use num_bigint::BigInt;
 
 static WAITED_FOR_EMPTY: Element = Element::Bytes(Bytes(Cow::Borrowed(b"WAITED_FOR_EMPTY")));
 static RESULT_NOT_PRESENT: Element = Element::Bytes(Bytes(Cow::Borrowed(b"RESULT_NOT_PRESENT")));
@@ -715,7 +716,11 @@ impl StackMachine {
                 let b = self.pop_element().await;
                 debug!("sub {:?} - {:?}", a, b);
                 let c = match (a, b) {
-                    (Element::Int(a), Element::Int(b)) => Element::Int(a - b),
+                    // checking for possible overflow
+                    (Element::Int(a), Element::Int(b)) => match a.checked_sub(b) {
+                        None => Element::BigInt(BigInt::from(a) - BigInt::from(b)),
+                        Some(c) => Element::Int(c),
+                    },
                     (Element::Float(a), Element::Float(b)) => Element::Float(a - b),
                     (Element::Double(a), Element::Double(b)) => Element::Double(a - b),
                     (Element::BigInt(a), Element::BigInt(b)) => Element::BigInt(a - b),

--- a/scripts/run_bindingtester.sh
+++ b/scripts/run_bindingtester.sh
@@ -6,9 +6,11 @@ fdb_rs_dir=$(pwd)
 fdb_builddir=${fdb_rs_dir:?}/target/foundationdb_build
 cd "${fdb_builddir:?}/foundationdb"
 
+
 # These are faulty seeds, for now, it is good to check them all the time to avoid regression.
 ./bindings/bindingtester/bindingtester.py --num-ops 1000 --api-version 630 --test-name api --compare python rust --seed 3534790651
 ./bindings/bindingtester/bindingtester.py --num-ops 1000 --api-version 630 --test-name api --compare python rust --seed 3864917676
+./bindings/bindingtester/bindingtester.py --num-ops 1000 --api-version 630 --test-name api --concurrency 5 rust --seed 3153055325
 
 START=1
 END=${1-1}

